apps/dashboard/src/components/variable/edit-variable-popover.tsx

```diff
@@ -20,20 +49,333 @@ export const EditVariablePopover = ({
+        return false;
+      }
+
+      setVariableError('');
+      return true;
+    },
+    [variable, isAllowedVariable]
+  );
+
+  const validateVariableDebounced = useDebounce(validateVariable, 2000);
+
+  // Set initial test value when popover opens
+  const handlePopoverOpen = useCallback(() => {
+    track(TelemetryEvent.VARIABLE_POPOVER_OPENED);
+
+    // Set a default test value based on the first filter, if any
+    if (filters.length > 0) {
+      const firstFilter = filters[0];
+      const sampleValue = getDefaultSampleValue(firstFilter.value, isEnhancedDigestEnabled);
+
+      if (sampleValue) {
+        setPreviewValue(sampleValue);
+      }
+    } else {
+      // Default to a simple string if no filters
+      setPreviewValue('Hello World');
+    }
+  }, [filters, track, isEnhancedDigestEnabled]);
+
+  const handleNameChange = useCallback(
+    (newName: string) => {
+      setName(newName);
+      validateVariableDebounced(newName);
+    },
+    [setName, validateVariableDebounced]
+  );
+
+  const handleDefaultValueChange = useCallback(
+    (newDefaultVal: string) => {
+      setDefaultVal(newDefaultVal);
+    },
+    [setDefaultVal]
+  );
+
+  const handleRawLiquidChange = useCallback(
+    (value: string) => {
+      const { parsedName, parsedDefaultValue } = parseRawInput(value);
+      setName(parsedName);
+      setDefaultVal(parsedDefaultValue);
+    },
+    [parseRawInput, setName, setDefaultVal]
+  );
+
+  const { handleReorder, handleFilterToggle, handleParamChange, getFilteredFilters } = useFilterManager({
+    initialFilters: filters,
+    onUpdate: setFilters,
+  });
+
+  const suggestedFilters = useSuggestedFilters(name, filters);
+  const filteredFilters = useMemo(() => getFilteredFilters(searchQuery), [getFilteredFilters, searchQuery]);
+
+  const currentLiquidValue = useMemo(
+    () => originalVariable || formatLiquidVariable(name, defaultVal, filters, isEnhancedDigestEnabled),
+    [originalVariable, name, defaultVal, filters, isEnhancedDigestEnabled]
+  );
+
+  const handleOpenChange = useCallback(
+    (open: boolean) => {
+      if (!open && !validateVariable(name)) {
+        return;
+      }
+
+      const newValue = formatLiquidVariable(name, defaultVal, filters, isEnhancedDigestEnabled);
+
+      if (!open) {
+        track(TelemetryEvent.VARIABLE_POPOVER_APPLIED, {
+          variableName: name,
+          hasDefaultValue: !!defaultVal,
+          filtersCount: filters.length,
+          filters: filters.map((filter) => filter.value),
+        });
+        setVariableError('');
+        onUpdate(newValue);
+      }
+
+      onOpenChange(open, newValue);
+    },
+    [validateVariable, onOpenChange, name, defaultVal, filters, track, onUpdate, isEnhancedDigestEnabled]
+  );
+
+  const handleClosePopover = useCallback(() => {
+    handleOpenChange(false);
+  }, [handleOpenChange]);
+
+  useEscapeKeyManager(id, handleClosePopover, EscapeKeyManagerPriority.POPOVER, open);
```

Comment:
managing the order of clicked esc buttons: first it will close the popover, then the sidebar; currently it closes sidebar on prod